File: 2018_Graduation_project#convert_to_java/TestApp/app/src/main/java/com/android/test/bitmap_rotate.java
package com.android.test;


import android.graphics.Matrix;
import android.graphics.Bitmap;

public class bitmap_rotate {
    public static Bitmap Rotate(Bitmap source, int degree) {
        // Initialize a new matrix
        Matrix matrix = new Matrix();
        // Rotate the bitmap
        matrix.postRotate(((float) degree));
        return Bitmap.createBitmap(source, 0, 0, source.getWidth(), source.getHeight(), matrix, true);
    }
}

File: 2018_Graduation_project#android&kotlin/app/src/main/java/com/example/master/a20180609_android_kotlin/bitmap_rotate.kt
package com.example.master.a20180609_android_kotlin

import android.graphics.Bitmap
import android.graphics.Matrix

fun Bitmap.rotate(degree:Int): Bitmap {
    // Initialize a new matrix
    val matrix = Matrix()

    // Rotate the bitmap
    matrix.postRotate(degree.toFloat())

    // Resize the bitmap
    val scaledBitmap = Bitmap.createScaledBitmap(
            this,
            width,
            height,
            true
    )

    // Create and return the rotated bitmap
    return Bitmap.createBitmap(
            scaledBitmap,
            0,
            0,
            scaledBitmap.width,
            scaledBitmap.height,
            matrix,
            true
    )
}